src/apps/bristleback_apps/pme_do_sensor/user_code/pme_do_sensor.cpp

```diff
@@ -185,13 +185,22 @@ bool PmeSensor::getDoData(PmeDissolvedOxygenMsg::Data &d) {
           q_signal.type != TYPE_DOUBLE) {
         printf("!  Parsed invalid DOT data");
       } else {
+        d.header.version = PmeDissolvedOxygenMsg::VERSION;
         d.header.reading_time_utc_ms = rtcGetMicroSeconds(&time_and_date) / 1000;
         d.header.reading_uptime_millis = uptimeGetMs();
+        d.header.sensor_reading_time_ms = 0;
         d.temperature_deg_c = temp_signal.data.double_val;
-        d.do_mg_per_l = do_signal.data.double_val;
+        d.do_mg_per_l =
+            do_signal.data.double_val * salinityFactor(d.temperature_deg_c, salinityPpt);
         d.quality = q_signal.data.double_val;
-        // DO Sat% not available at this time; setting to NULL causes error (converting NULL to double)
-        //d.do_saturation_pct = 0;
+        if (_is_baro_valid) {
+          d.do_saturation_pct = do_signal.data.double_val /
+                                doConcMg(d.temperature_deg_c, _barometric_pressure_mbar, 0.0) *
+                                100.0;
+        } else {
+          d.do_saturation_pct = 0.0;
+        }
```

Comment:
@b-haist please verify this saturation calculation is correct.

Reply:
Peter said the calculation looks good, but we decided to change the invalid value from 0.0 to NAN.